### PR TITLE
fix: osh.pl: validate user and host format

### PR DESCRIPTION
**Pull request by speed47**

### CODE REVIEW

File: bin/shell/osh.pl [hunk 1]
```diff
@@ -547,6 +547,13 @@ sub main_exit {
     }
 }
 
+if (!OVH::Bastion::is_valid_remote_user(user => $user)) {
+    main_exit OVH::Bastion::EXIT_INVALID_REMOTE_USER, 'invalid_remote_user', "Remote user name '$user' seems invalid";
+}
+if ($host !~ m{^[a-z0-9._-]+$}i) {
```
Comment (deathiop) [inline bin/shell/osh.pl:553]:
```suggestion
if ($host !~ m{^[a-zA-Z0-9._-]+$}) {
```

**Review by deathiop — CHANGES_REQUESTED** [review]
Couldn't it be tested?

File: bin/shell/osh.pl (hunk 1 above)
Comment (speed47) [reply bin/shell/osh.pl]: Thx, applied

**Comment by speed47** [comment]
> Couldn't it be tested?

added tests

**Review by Alkorin — DISMISSED** [review]

**Comment by speed47** [comment]
Gah, we get a 403 when fetching the latest ttyrec release from the API... probably rate limiting. Hmm.

**Review by Alkorin — DISMISSED** [review]

**Review by deathiop — APPROVED** [review]

**Pull request merged**